src/go/configmanager/config_manager.go

```diff
@@ -121,31 +120,18 @@ func NewConfigManager(mf *metadata.MetadataFetcher, opts options.ConfigGenerator
 		return nil, fmt.Errorf(`failed to set rollout strategy. It must be either "managed" or "fixed"`)
 	}
 
-	// Create secured http client with rootCertsPath.
-	if serviceConfigFetcherClient, err = newServiceConfigFetcherClient(time.Duration(*commonflags.HttpRequestTimeoutS) * time.Second); err != nil {
-		return nil, fmt.Errorf(`failed to create https client to call ServiceManagement service, got error: %v`, err)
-	}
-
 	if rolloutStrategy == util.ManagedRolloutStrategy {
-		// try to fetch rollouts and get newest config, if failed, NewConfigManager exits with failure
-		newRolloutID, newConfigID, err := loadConfigFromRollouts(m.serviceName, m.curRolloutID, m.curConfigID, mf)
-		if err != nil {
-			return nil, err
-		}
-		if m.curRolloutID != newRolloutID && m.curConfigID != newConfigID {
-			m.curRolloutID = newRolloutID
-			m.curConfigID = newConfigID
-			if err := m.updateSnapshot(); err != nil {
-				return nil, err
-			}
+		if m.serviceConfigFetcher, err = sc.NewServiceConfigFetcher(mf, opts, m.serviceName, ""); err != nil {
+			return nil, fmt.Errorf(`failed to create https client to call ServiceManagement service, got error: %v`, err)
+
 		}
 	} else {
 		// rollout strategy is fixed mode
-		configID := *ServiceConfigID
-		if configID == "" {
+		configId := *ServiceConfigID
+		if configId == "" {
 			if checkMetadata && mf != nil {
-				configID, err = mf.FetchConfigId()
-				if configID == "" || err != nil {
+				configId, err = mf.FetchConfigId()
```

Comment:
I don't fully understand, where is Metadata Fetcher getting the config id from? Is it a custom attribute attached to the VM? cc @qiwzhang, is this for ESPv1 compatibility?